includes/Modules/Search_Console.php

```diff
@@ -423,6 +416,52 @@ function( $dataset ) {
 		return (bool) $has_data;
 	}
 
+	/**
+	 * Gets the property ID.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string Property ID URL if set, or empty string.
+	 */
+	protected function get_property_id() {
+		$option = $this->get_settings()->get();
+
+		return $option['propertyID'];
+	}
+
+	/**
+	 * Detects the property ID to use for this site.
+	 *
+	 * This method runs a Search Console API request. The determined ID should therefore be stored and accessed through
+	 * {@see Search_Console::get_property_id()} instead.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string Property ID, or empty string if none found.
+	 */
+	protected function detect_property_id() {
+		$properties = $this->get_data( 'matched-sites' );
+		if ( is_wp_error( $properties ) || ! $properties ) {
+			return '';
+		}
+
+		// If there are multiple, prefer URL property over domain property.
+		if ( count( $properties ) > 1 ) {
+			$url_properties = array_filter(
+				$properties,
+				function( $property ) {
+					return 0 !== strpos( $property['siteURL'], 'sc-domain:' );
+				}
+			);
+			if ( count( $url_properties ) > 0 ) {
+				$properties = $url_properties;
+			}
+		}
+
+		$property = array_shift( $properties );
+		return $property['siteURL'];
```

Comment:
This looks like it might raise an error if there were no properties found since it would be trying to `false['siteURL']`

Reply:
This should already be caught on top of the method by `! $properties`.